src/net.cpp

```diff
@@ -2861,3 +2861,23 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& ad) const
 
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
+
+void LogMessage(const CNode& node, const std::string& msg_type, const Span<const unsigned char>& data, bool is_incoming)
+{
+    auto time = GetTime<std::chrono::microseconds>();
+
+    fs::path base_path = GetDataDir() / "peer_logging" / node.addr.ToString();
+    fs::create_directories(base_path);
+
+    fs::path path = base_path / (is_incoming ? "msgs_recv.dat" : "msgs_sent.dat");
+    CAutoFile f(fsbridge::fopen(path, "ab"), SER_DISK, CLIENT_VERSION);
+
+    f.write((const char*)&time, sizeof(time));
+    f.write(msg_type.data(), msg_type.length());
+    for (auto i = msg_type.length(); i < CMessageHeader::COMMAND_SIZE; ++i) {
+        f << '\0';
+    }
+    uint32_t size = data.size();
+    f.write((const char*)&size, sizeof(size));
```

Comment:
Same as above, this should be guaranteed to be written out in little endian byte order.

Reply:
done